src/charm.py

```diff
@@ -294,6 +324,40 @@ def reload_pgbouncer(self) -> None:
 
         self.check_pgb_running()
 
+    def _generate_monitoring_service(self, enabled: bool = True) -> Dict[str, str]:
+        if enabled:
+            stats_password = self.peers.get_secret("app", MONITORING_PASSWORD_KEY)
+            command = (
+                f'pgbouncer_exporter --web.listen-address=:{METRICS_PORT} --pgBouncer.connectionString="'
+                f'postgres://{self.backend.stats_user}:{stats_password}@localhost:{self.config["listen_port"]}/pgbouncer?sslmode=disable"'
```

Comment:
Should we be using TLS here if enabled? It is a local service, so it shouldn't matter.

Reply:
From what Mykola told me sometime ago, all the connections to the database should use TLS when available, but I think all the tools in the PG charm are using IP addresses on VM and hostnames on K8S, so I'm not sure whether it's needed for localhost. It looks that it's not worth locally.